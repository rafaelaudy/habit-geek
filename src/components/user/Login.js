import React, { useState } from "react";
import { navigate } from "@reach/router";
import { useTranslation } from "react-i18next";
import * as userService from "@rafael.audy/habit-geek-utils/services/userService";
import UserForm from "./UserForm";

const Login = ({ saveSession }) => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [validityChecked, setValidityChecked] = useState(false);
  const { t } = useTranslation();

  const login = event => {
    event.preventDefault();
    setValidityChecked(true);

    if (event.target.checkValidity()) {
      userService
        .login(email, password)
        .then(saveSession)
        .catch();
    }
  };

  return (
    <div className="mobile-size-container">
      <form
        className={`${validityChecked ? "was-validated" : ""}`}
        onSubmit={login}
        noValidate
      >
        <h2>{t("login-title")}</h2>

        <UserForm
          email={email}
          password={password}
          setEmail={setEmail}
          setPassword={setPassword}
        ></UserForm>

        <hr className="mb-4" />
        <button type="submit" className="btn btn-primary btn-lg btn-block">
          {t("login-login")}
        </button>
      </form>
      <div>
        {t("login-sign-up-label")}
        <button
          onClick={() => navigate("/habit-geek/signup")}
          className="px-0 btn btn-link"
        >
          {t("login-sign-up")}
        </button>
      </div>
      <div>
        {t("login-forgot-password-label")}
        <button
          onClick={() => navigate("/habit-geek/forgotpassword")}
          className="px-0 btn btn-link"
        >
          {t("login-forgot-password")}
        </button>
      </div>
    </div>
  );
};

export default Login;
