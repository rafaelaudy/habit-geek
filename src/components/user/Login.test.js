import React from "react";
import { shallow } from "enzyme";
import Login from "./Login";

//TODO
describe("Login component", () => {
  it("Renders static elements", () => {
    const login = shallow(<Login />);
    expect(login).toMatchSnapshot();
  });
});
