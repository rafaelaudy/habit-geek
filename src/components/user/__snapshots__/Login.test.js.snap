// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component Renders static elements 1`] = `
<div
  className="mobile-size-container"
>
  <form
    className=""
    noValidate={true}
    onSubmit={[Function]}
  >
    <h2>
      login-title
    </h2>
    <UserForm
      email=""
      password=""
      setEmail={[Function]}
      setPassword={[Function]}
    />
    <hr
      className="mb-4"
    />
    <button
      className="btn btn-primary btn-lg btn-block"
      type="submit"
    >
      login-login
    </button>
  </form>
  <div>
    login-sign-up-label
    <button
      className="px-0 btn btn-link"
      onClick={[Function]}
    >
      login-sign-up
    </button>
  </div>
  <div>
    login-forgot-password-label
    <button
      className="px-0 btn btn-link"
      onClick={[Function]}
    >
      login-forgot-password
    </button>
  </div>
</div>
`;
